code/modules/mob/mob.dm

```diff
@@ -689,41 +689,34 @@
 /mob/proc/can_stand_overridden()
 	return 0
 
-//Updates lying and icons
-/mob/proc/update_lying()
-	if(!resting && cannot_stand() && can_stand_overridden())
-		lying = FALSE
-	else if(buckled)
-		anchored = TRUE
-		if(istype(buckled))
-			if(buckled.buckle_lying == -1)
-				lying = incapacitated(INCAPACITATION_KNOCKDOWN)
-			else
-				lying = buckled.buckle_lying
-			if(buckled.buckle_movable)
-				anchored = FALSE
-	else
-		lying = incapacitated(INCAPACITATION_KNOCKDOWN)
-
-/mob/proc/UpdateLyingBuckledAndVerbStatus()
-	var/last_lying = lying
-	update_lying()
-	if(buckled)
-		anchored = (!istype(buckled) || !buckled.buckle_movable)
-	if(lying)
-		set_density(0)
-		drop_held_items()
+//Updates lying, transform and icons
+/mob/proc/update_posture(force_update)
+
+	var/list/available_postures = get_available_postures()
+	if(length(available_postures) <= 0)
+		return // No postures, no point doing any of this.
+
+	if(length(available_postures) == 1)
+		// If we only have one posture, use that.
+		. = set_posture(available_postures[1], skip_buckled_update = TRUE)
+	else if(istype(buckled) && buckled.buckle_lying != -1)
+		// If we're buckled to something that forces a posture, use that.
+		. = set_posture(buckled.buckle_lying ? /decl/posture/lying : /decl/posture/standing, skip_buckled_update = TRUE)
+	else if(incapacitated(INCAPACITATION_KNOCKDOWN) || (cannot_stand() && !can_stand_overridden()))
+		// If we're straight up knocked over, set that.
+		if(!current_posture.prone)
+			. = set_posture(/decl/posture/lying, skip_buckled_update = TRUE)
+	else if(!current_posture.deliberate)
+		// If we're not deliberately lying, and we can stand, stand up.
+		. = set_posture(/decl/posture/standing, skip_buckled_update = TRUE)
 	else
-		set_density(initial(density))
+		. = FALSE
+
+	anchored = buckled ? (!istype(buckled) || !buckled.buckle_movable) : initial(anchored)
 	reset_layer()
 
-	//Temporarily moved here from the various life() procs
-	//I'm fixing stuff incrementally so this will likely find a better home.
-	//It just makes sense for now. ~Carn
-	if( update_icon )	//forces a full overlay update
-		update_icon = 0
```

Comment:
hold on, *this* is where `mob.update_icon` was handled? was that moved or was the var just deprecated